### PR TITLE
`Roles | Bugfix`: Typing Errors

### PR DESCRIPTION
**Pull request by Da-Colon**
- Was getting an error when Role details component would mount. Discovered that `linearStream.depositAmount` was coming in from subgraph as a string.
- Then I got an error about `payment.amount.bigIntValue` being undefined, added `?`
- Then I got an error about a `getTime()`

### CODE REVIEW

File: src/components/pages/Roles/RolePaymentDetails.tsx [hunk 1]
```diff
@@ -121,7 +121,7 @@ export function RolePaymentDetails({
   }, [addressPrefix, navigate, safe?.address, withdraw]);
 
   const amountPerWeek = useMemo(() => {
-    if (!payment.amount.bigintValue) {
+    if (!payment.amount?.bigintValue) {
```
Comment (adamgall) [inline src/components/pages/Roles/RolePaymentDetails.tsx:124]: If `amount` might be undefined, it should be typed as so, right?

File: src/components/pages/Roles/RolePaymentDetails.tsx [hunk 2]
```diff
@@ -148,10 +148,15 @@ export function RolePaymentDetails({
   }, [payment.amount, payment.asset?.address, assetsFungible]);
 
   const start =
-    payment.cliffDate === undefined ? payment.startDate.getTime() : payment.cliffDate.getTime();
-  const end = payment.endDate.getTime();
+    payment.cliffDate === undefined && payment.startDate !== undefined
```
Comment (adamgall) [inline src/components/pages/Roles/RolePaymentDetails.tsx:151]: If the `startDate` might be undefined, it should be typed as so, right?

File: src/components/pages/Roles/RolePaymentDetails.tsx [hunk 3]
```diff
@@ -148,10 +148,15 @@ export function RolePaymentDetails({
...
+      ? payment.startDate.getTime()
+      : payment.cliffDate !== undefined
+        ? payment.cliffDate.getTime()
+        : undefined;
+
+  const end = payment.endDate ? payment.endDate.getTime() : undefined;
```
Comment (adamgall) [inline src/components/pages/Roles/RolePaymentDetails.tsx:157]: If the `endDate` might be undefined, it should be typed as so, right?

**Comment by Da-Colon** [comment]
@adamgall Yeeeeep My next PR takes care of that. (which I believe you've glanced at already). Working on these fixes showed that exact problem. This PR makes things work. Next PR updates the types and fixes all the other spots that are ticking tie bombs

**Review by DarksightKellar — APPROVED** [review]

**Review by adamgall — APPROVED** [review]
Approving due to @Da-Colon's last comment

File: src/components/pages/Roles/RolePaymentDetails.tsx (hunk 3 above)
Comment (adamgall) [reply src/components/pages/Roles/RolePaymentDetails.tsx:157]: https://github.com/decentdao/decent-interface/pull/2262#pullrequestreview-2245430499

File: src/components/pages/Roles/RolePaymentDetails.tsx (hunk 2 above)
Comment (adamgall) [reply src/components/pages/Roles/RolePaymentDetails.tsx:151]: https://github.com/decentdao/decent-interface/pull/2262#pullrequestreview-2245430499

File: src/components/pages/Roles/RolePaymentDetails.tsx (hunk 1 above)
Comment (adamgall) [reply src/components/pages/Roles/RolePaymentDetails.tsx:124]: https://github.com/decentdao/decent-interface/pull/2262#pullrequestreview-2245430499

**Pull request merged**